### PR TITLE
fix(AFHDS3): V0 ibus option and failsafe malfunction

### LINKED ISSUE #3791: AFHDS3: FGr4P receiver cannot setup iBUS IN
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Setting iBUS IN is not working properly, sensors readings not appeared in radio.

### Expected Behavior

After setup iBUS IN,  sensors readings should be able to display in radio.

### Steps To Reproduce

1. Setup iBUS IN
2. No sensor reading in radio

### Version

2.9.0

### Transmitter

Flysky EL18

### Operating System (OS)

_No response_

### OS Version

_No response_

### Anything else?

_No response_

### LINKED ISSUE #3792: AFHDS3: Failsafe broken
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Failsafe is not updated in receiver when changing the settings in the radio.

### Expected Behavior

Failsafe should be able to update properly.

### Steps To Reproduce

1. Change failsafe settings
2. Test failsafe
3. Failed

### Version

2.9.0

### Transmitter

Flysky EL18

### Operating System (OS)

_No response_

### OS Version

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by richardclli**
Fixes #3791 
Fixes #3792

Summary of changes:

### CODE REVIEW

File: radio/src/pulses/afhds3.cpp [hunk 1]
```diff
@@ -874,49 +874,71 @@ bool ProtoState::syncSettings()
...
+    data[37] = (cfg->v1.PWMFrequenciesV1.Synchronized>>24) & 0xff;
     trsp.putFrame(COMMAND::SEND_COMMAND, FRAME_TYPE::REQUEST_SET_EXPECT_DATA, data, sizeof(data));
     return true;
   }
-  if (checkDirtyFlag(DC_RX_CMD_BUS_TYPE_V0))  
+  if (checkDirtyFlag(DC_RX_CMD_BUS_TYPE_V0))
   {
     TRACE("AFHDS3 [RX_CMD_BUS_TYPE_V0]");
-    uint8_t data[] = { (uint8_t)(RX_CMD_BUS_TYPE_V0&0xFF), (uint8_t)((RX_CMD_BUS_TYPE_V0>>8)&0xFF), 1, cfg->others.ExternalBusType };
+    if(1>=cfg->others.ExternalBusType) //IBUS1
```
Comment (raphaelcoeffic) [inline radio/src/pulses/afhds3.cpp:920]: No magic numbers :-D

File: radio/src/pulses/afhds3_config.h [hunk 2]
```diff
@@ -129,7 +129,8 @@ PACK(struct sDATA_ConfigV1 {
 
 PACK(struct sDATA_Others {
   uint8_t buffer[sizeof(sDATA_ConfigV1)];
-  uint8_t ExternalBusType;  // eEB_BusType
+  uint8_t ExternalBusType;  // eEB_BusType IBUS1:0;IBUS2:1(Not supported yet);SBUS:2
+  uint8_t iBusType;  //IBUS-OUT:0; IBUS-IN:1
```
Comment (raphaelcoeffic) [inline radio/src/pulses/afhds3_config.h:133]: Let's use a clean `enum` for these 2.

Comment (gagarinlg) [reply radio/src/pulses/afhds3_config.h:133]: with
```suggestion
  enum IBusType: uint8_t {IBusOut = 0, IBusIn = 1}:
```
you can retain the same size of the data struct

Comment (richardclli) [reply radio/src/pulses/afhds3_config.h]: I have made some changes.  See if this is good to you all.

**Comment by richardclli** [comment]
@pfeerick I think this is good for merge, for now.  There are some historical problem about V0 receivers and need more time for Flysky to resolve in the TX firmware first.

**Comment by pfeerick** [comment]
Sounds good to me... if there are no further concerns from @raphaelcoeffic @gagarinlg I'll review and try this shortly.

**Review by pfeerick — APPROVED** [review]
Tested on FTr12B w/ EL18, both iBUS IN and failsafe settings now runtime configurable rather than bind time (or perhaps not at all in some instances).

**Comment by philmoz** [comment]
I think line 924 is wrong (compiler complains).
```
      cfg->others.ExternalBusType == EB_BT_IBUS1_OUT;
```

should probably be assignment not comparison.
```
      cfg->others.ExternalBusType = EB_BT_IBUS1_OUT;
```

**Comment by pfeerick** [comment]
@richardclli I haven't pushed the commit for this on the 2.9 branch yet, so can make that change before pushing it if that is the only change required

**Comment by richardclli** [comment]
> @richardclli I haven't pushed the commit for this on the 2.9 branch yet, so can make that change before pushing it if that is the only change required
> 
> edit: ouch... managed to misattribute another commit to this PR ... bad VSCode!

Yes, that is the only change needed.

**Pull request merged**